homeassistant/components/light/tellstick.py

```diff
@@ -57,9 +56,8 @@ def _parse_ha_data(self, kwargs):
 
     def _parse_tellcore_data(self, tellcore_data):
         """Turn the value received from tellcore into something useful."""
-        if tellcore_data is not None:
-            brightness = int(tellcore_data)
-            return brightness
+        if tellcore_data:
```

Comment:
This works, but the fix is conceptually in the wrong place. When we get into `_update_model_from_command`, we know the `tellcore_command`, and we know that it's `TELLSTICK_DIM`. That's the only use-case where parsing the data makes sense, per the [docs](http://developer.telldus.se/doxygen/group__core.html#ga9f3a4c489b9ef35e41b07595b4185ecc):

> If method is TELLSTICK_DIM this holds the current value as a human readable string, example "128" for 50%.

The fix should be in `_update_model_from_command`, never calling `_parse_tellcore_data`, which by the time is called doesn't know what the command was, and can't reason about whether it's a failure that we have no data or if it's due to receiving one of the commands that have no data in the first place.